pkg/apis/pipeline/v1alpha1/pipelineparams_validation.go

```diff
@@ -0,0 +1,107 @@
+/*
+Copyright 2018 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	"fmt"
+	"net/url"
+
+	"github.com/knative/pkg/apis"
+	"k8s.io/apimachinery/pkg/api/equality"
+)
+
+func (p *PipelineParams) Validate() *apis.FieldError {
+	if err := validateObjectMetadata(p.GetObjectMeta()); err != nil {
+		return err.ViaField("metadata")
+	}
+	return nil
+}
+
+func (ps *PipelineParamsSpec) Validate() *apis.FieldError {
+	if equality.Semantic.DeepEqual(ps, &PipelineParams{}) {
+		return apis.ErrMissingField(apis.CurrentField)
+	}
+	// PipelineParams must specify a service account to run under
+	if ps.ServiceAccount == "" {
+		return apis.ErrMissingField("pipelineparamsspec.ServiceAccount")
+	}
```

Comment:
i _think_ that service account should be optional - just in case a pipeline/task doesn't actually need to provide it. our integration tests are currently getting by without it (and on a related note, how are the integration tests passing without it on this branch???)